pkg/client/introduce/client.go

```diff
@@ -53,176 +38,63 @@ func New(ctx Provider, inv *didexchange.Invitation) (*Client, error) {
 		return nil, errors.New("cast service to Introduce Service failed")
 	}
 
-	store, err := ctx.StorageProvider().OpenStore(introduce.Introduce)
-	if err != nil {
-		return nil, err
-	}
-
 	return &Client{
-		Event:      introduceSvc,
-		service:    introduceSvc,
-		store:      store,
-		defaultInv: inv,
-		newUUID:    func() string { return uuid.New().String() },
+		Event:   introduceSvc,
+		service: introduceSvc,
 	}, nil
 }
 
-// InvitationEnvelope is a helper function that returns the dependency needed for the
-// service to proceed with the protocol. Dependency should be passed to the service through `Continue` function.
-// The function should never return an error. Instead of an error we provide a callable interface
-// and a state machine that will act according to the provided data.
-// Dependency is populated after executing the following functions:
-//  - SendProposal
-//  - SendProposalWithInvitation
-//  - HandleRequest
-//  - HandleRequestWithInvitation
-// usage: e.Continue(c.InvitationEnvelope(threadID))
-func (c *Client) InvitationEnvelope(thID string) *InvitationEnvelope {
-	opts, err := c.getInvitationEnvelope(thID)
-
-	if errors.Is(err, storage.ErrDataNotFound) {
-		return &InvitationEnvelope{Inv: c.defaultInv}
-	}
+// SendProposal sends a proposal to the introducees (the client does not have a public Invitation).
+func (c *Client) SendProposal(recipient1, recipient2 *introduce.Recipient) error {
+	proposal1 := introduce.CreateProposal(recipient1.To)
+	proposal2 := introduce.CreateProposal(recipient2.To)
+
+	introduce.WrapWithMetadataPIID(proposal1, proposal2)
 
+	_, err := c.service.HandleOutbound(proposal1, recipient1.MyDID, recipient1.TheirDID)
 	if err != nil {
-		logger.Errorf("invitation envelope: %v", err)
-		return &InvitationEnvelope{}
+		return fmt.Errorf("handle outbound: %w", err)
 	}
 
-	return opts
-}
+	_, err = c.service.HandleOutbound(proposal2, recipient2.MyDID, recipient2.TheirDID)
 
-// SendProposal sends a proposal to the introducees (the client does not have a public Invitation).
-func (c *Client) SendProposal(recipient1, recipient2 *introduce.Recipient) error {
-	return c.sendProposal(InvitationEnvelope{
-		Recps: []*introduce.Recipient{recipient1, recipient2},
-	})
+	return err
 }
 
 // SendProposalWithInvitation sends a proposal to the introducee (the client has a public Invitation).
 func (c *Client) SendProposalWithInvitation(inv *didexchange.Invitation, recipient *introduce.Recipient) error {
-	return c.sendProposal(InvitationEnvelope{
-		Inv:   inv,
-		Recps: []*introduce.Recipient{recipient},
-	})
+	proposal := introduce.CreateProposal(recipient.To)
+
+	introduce.WrapWithMetadataPublicInvitation(proposal, inv)
+
+	_, err := c.service.HandleOutbound(proposal, recipient.MyDID, recipient.TheirDID)
```

Comment:
what is the ignored return value?

Reply:
It is PIID, I did it similar to did-exchange (connectionID). But we do not use it. Not even sure whether it will be used.

Reply:
ok